js/main.js

```diff
@@ -0,0 +1,38 @@
+var $form = document.querySelector('#entry-area');
+var $photo = document.querySelector('#input-img');
+var $photoURL = document.querySelector('#photo-url');
+
+function updatePhotoPreview(event) {
+  // test image (pikachu doing a ✌): https://i.pinimg.com/736x/5c/17/aa/5c17aa712336ec66a892e290f4504a44.jpg
+  if ($form.elements.photoURL.value !== '') {
+    $photo.setAttribute('src', $form.elements.photoURL.value);
+    $photo.setAttribute('alt', 'User selected image');
+  } else {
+    $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  }
+}
+
+function saveFormData(event) {
+  event.preventDefault();
+
+  var formData = {
+    entryId: data.nextEntryId,
+    title: $form.elements.entryTitle.value,
+    photoURL: $form.elements.photoURL.value,
+    notes: $form.elements.entryNotes.value
+  };
+  data.entries.unshift(formData);
+  data.nextEntryId++;
+  // reset form
+  $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $photo.setAttribute('alt', 'placeholder image');
+  $form.reset();
+}
+
+function persistFormData(event) {
+  localStorage.setItem('code-journal-entry-data', JSON.stringify(data));
+}
```

Comment:
You should move all localStorage-related code into `data.js` so that it is in the same file as the data it is preserving.